build/LocalizationTasks/EmitLocalizedResources.cs

```diff
@@ -0,0 +1,112 @@
+        private static readonly string LogicalNameMetadata = "LogicalName";
+
+        [Required]
+        public ITaskItem[] NeutralResources { get; set; }
+
+        /// <summary>
+        /// Name of the main assembly
+        /// </summary>
+        [Required]
+        public string AssemblyName { get; set; }
+        
+        /// <summary>
+        /// Directory path where localized resx should be generated in 
+        /// </summary>
+        [Required]
+        public string LocalizedResxRoot { get; set; }
+
+        /// <summary>
+        /// Path to the root directory that contains the xlf files for a neutral resource
+        /// Assumes all neutral resource have the same relative path to their xlf
+        /// </summary>
+        [Required]
+        public string RelativePathToXlfRoot { get; set; }
+
+        [Output]
+        public ITaskItem[] ResolvedXlfResources { get; set; }
+
+        [Output]
+        public ITaskItem[] ResolvedLocalizedResxResources { get; set; }
+
+        public override bool Execute()
+        {
+            if (!NeutralResources.Any())
+            {
+                Log.LogMessage("No neutral resources found");
+                return true;
+            }
+
+            ResolvedXlfResources = NeutralResources.SelectMany(ComputeXlfResources).ToArray();
+
+            ResolvedLocalizedResxResources = ResolvedXlfResources.Select(ComputeLocalizedResource).ToArray();
+
+            return true;
+        }
+
+        private ITaskItem ComputeLocalizedResource(ITaskItem xlf)
+        {
+            var resxName = xlf.GetMetadata("FileName");
+            var resxFileName = resxName + ".resx";
+            var resxPath = Path.Combine(LocalizedResxRoot, resxFileName);
+
+            var resxItem = new TaskItem(resxPath);
+            resxItem.SetMetadata(NeutralResxMetadata, xlf.GetMetadata(NeutralResxMetadata));
+            resxItem.SetMetadata(ParentXlfMetadata, xlf.ItemSpec);
+            resxItem.SetMetadata(ComputedCultureCodeMetadata, xlf.GetMetadata(ComputedCultureCodeMetadata));
+            resxItem.SetMetadata(LogicalNameMetadata, $"{AssemblyName}.{resxName}.resources");
+
+            xlf.SetMetadata(ChildResxMetadata, resxPath);
+
+            return resxItem;
+        }
+
+        private IEnumerable<ITaskItem> ComputeXlfResources(ITaskItem neutralResource)
+        {
+            var neutralResxRootDirectory = Path.GetDirectoryName(neutralResource.GetMetadata("FullPath"));
+            var xlfRootPath = Path.Combine(neutralResxRootDirectory, RelativePathToXlfRoot);
+
+            // todo: throw if xlf root does not exist; throw in other places too
+
+            var resx = neutralResource.ItemSpec;
+
+            return (Directory.EnumerateFiles(xlfRootPath).Where(f => IsValidXlf(f, resx)).Select((f => CreateXLFTaskItemForNeutralResx(f, resx))));            
+        }
+
+        private bool IsValidXlf(string xlfPath, string resx)
```

Comment:
Why is this needed? It seems a bit scary.

Actually the whole enumerate-files approach scares me a bit; I prefer an explicit-inputs approach.